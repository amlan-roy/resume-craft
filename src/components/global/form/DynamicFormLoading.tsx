import React from "react";
import Loading from "@/components/global/form/form-sections/Loading";

/*
Creating a separate component for loading because of an issue with react hook forms. 
The form data is fetch async, and if once the default data is set in the react hook form method inside dynamic form component, 
it is not updated once the default values data is updated. 
Hence we should not init the method until the data is final. 
Hence we are using a separate loading component to show the loading state. 
Can fix this later, but for now this is the temp solution.
*/
type DynamicFormLoadingProps = {};

const DynamicFormLoading: React.FC<DynamicFormLoadingProps> = () => {
  return (
    <div className="space-y-8">
      <section className="w-full flex flex-col gap-8">
        <Loading />
        <Loading />
      </section>
    </div>
  );
};
export default DynamicFormLoading;
