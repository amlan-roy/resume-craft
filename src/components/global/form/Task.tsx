import React from "react";
import { useSortable } from "@dnd-kit/sortable";
import { CSS } from "@dnd-kit/utilities";

type TaskProps = {
  id: Number;
  title: String;
};

const Task: React.FC<TaskProps> = ({ id, title }) => {
  const {
    attributes,
    listeners,
    setNodeRef,
    transform,
    transition,
    setActivatorNodeRef,
  } = useSortable({
    id: id.toString(),
  });

  const style = {
    transition,
    transform: CSS.Transform.toString(transform),
    touchAction: "none",
  };
  return (
    <div
      ref={setNodeRef}
      className="w-full p-2 border-2 border-solid border-black rounded-sm flex flex-row gap-2"
      style={style}
    >
      <div
        {...attributes}
        {...listeners}
        ref={setActivatorNodeRef}
        className="w-fit bg-slate-100"
      >
        ...
      </div>
      {title}
    </div>
  );
};
export default Task;
