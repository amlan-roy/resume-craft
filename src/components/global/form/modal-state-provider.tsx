"use client";

import React, { useCallback } from "react";

type ModalStateProviderProps = {
  children: React.ReactNode;
};

type ModalState = {
  title?: string;
  message?: string;
  isOpen: boolean;
  confirmText?: string;
  cancelText?: string;
  onConfirm?: () => void;
  onCancel?: () => void;
};

type ModalStateContextType = {
  modalState: ModalState;
  setModalState: (newState: ModalState) => void;
};

export const ModalStateContext = React.createContext<ModalStateContextType>({
  modalState: {
    isOpen: false,
  },
  setModalState: () => console.warn("No modal state provider"),
});

const ModalStateProvider: React.FC<ModalStateProviderProps> = ({
  children,
}) => {
  const [modalState, setModalState] = React.useState<ModalState>({
    isOpen: false,
  });

  const setModalStateHandler = useCallback((newState: ModalState) => {
    setModalState(() => newState);
  }, []);

  const modalStateContextValue = React.useMemo(() => {
    return {
      modalState,
      setModalState: setModalStateHandler,
    };
  }, [modalState, setModalStateHandler]);

  return (
    <ModalStateContext.Provider value={modalStateContextValue}>
      {children}
    </ModalStateContext.Provider>
  );
};

export default ModalStateProvider;
