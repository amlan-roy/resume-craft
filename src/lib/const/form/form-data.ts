import { z } from "zod";
import {
  SECTION,
  additionalSectionSchema,
  basicDetailsSectionSchema,
  educationSectionSchema,
  professionalSummarySectionSchema,
  projectsSectionSchema,
  skillsSectionSchema,
  workExperienceSectionSchema,
} from "@/lib/types/form";

export const DEFAULT_BASIC_DETAILS_FORM_VALUE: z.infer<
  typeof basicDetailsSectionSchema
> = {
  type: SECTION.BASIC_DETAILS,
  sectionTitle: "Basic Details",
  fields: {
    name: "",
  },
};

export const DEFAULT_PROFESSIONAL_SUMMARY_FORM_VALUE: z.infer<
  typeof professionalSummarySectionSchema
> = {
  type: SECTION.PROFESSIONAL_SUMMARY,
  sectionTitle: "Professional Summary",
  fields: {
    value: "",
  },
};

export const DEFAULT_WORK_EXPERIENCE_FORM_VALUE: z.infer<
  typeof workExperienceSectionSchema
> = {
  type: SECTION.WORK_EXPERIENCE,
  sectionTitle: "Work Experience",
  fields: [{ jobTitle: "", details: "", companyName: "", location: "" }],
};
export const DEFAULT_PROJECTS_FORM_VALUE: z.infer<
  typeof projectsSectionSchema
> = {
  type: SECTION.PROJECTS,
  sectionTitle: "Relevant Projects",
  fields: [
    {
      projectTitle: "",
      projectSubtitle: "",
      projectUrl: undefined,
      associatedWith: undefined,
      details: "",
    },
  ],
};
export const DEFAULT_SKILLS_FORM_VALUE: z.infer<typeof skillsSectionSchema> = {
  type: SECTION.SKILLS,
  sectionTitle: "Technical Skills",
  fields: [
    {
      skills: "",
    },
  ],
};
export const DEFAULT_EDUCATION_FORM_VALUE: z.infer<
  typeof educationSectionSchema
> = {
  type: SECTION.EDUCATION,
  sectionTitle: "Education",
  fields: [
    {
      universityName: "",
      degreeName: "",
      majorName: "",
      grade: "",
      location: "",
    },
  ],
};
export const DEFAULT_ADDITIONAL_FORM_VALUE: z.infer<
  typeof additionalSectionSchema
> = {
  type: SECTION.ADDITIONAL,
  sectionTitle: "Additional Information",
  fields: {
    value: "",
  },
};

export const DEFAULT_FORM_VALUE = {
  basicDetails: DEFAULT_BASIC_DETAILS_FORM_VALUE,
  optionalSections: [
    DEFAULT_PROFESSIONAL_SUMMARY_FORM_VALUE,
    DEFAULT_WORK_EXPERIENCE_FORM_VALUE,
    DEFAULT_PROJECTS_FORM_VALUE,
    DEFAULT_SKILLS_FORM_VALUE,
    DEFAULT_EDUCATION_FORM_VALUE,
    DEFAULT_ADDITIONAL_FORM_VALUE,
  ],
};
