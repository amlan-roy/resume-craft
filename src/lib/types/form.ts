import * as z from "zod";

export enum SECTION {
  ADDITIONAL = "ADDITIONAL",
  BASIC_DETAILS = "BASIC_DETAILS",
  EDUCATION = "EDUCATION",
  PROFESSIONAL_SUMMARY = "PROFESSIONAL_SUMMARY",
  PROJECTS = "PROJECTS",
  SKILLS = "SKILLS",
  WORK_EXPERIENCE = "WORK_EXPERIENCE",
}

export const durationFieldSchema = z
  .object({
    startDate: z.date().optional().nullish(),
    endDate: z.date().optional().nullish(),
    current: z.boolean().optional().nullish(),
  })
  .refine(
    (data) => {
      return !(data.endDate || !!data.current) || !!data.startDate;
    },
    (data) => {
      if (data.endDate) {
        return {
          message: "Start Date is required when you select an end date",
          path: ["startDate"],
        };
      }
      return {
        message: "Start Date is required when you select the current checkbox",
        path: ["startDate"],
      };
    }
  );

export type TDuration = {
  startDate?: string | null | undefined;
  endDate?: string | null | undefined;
  current?: boolean | null | undefined;
};

export const basicDetailsFieldSchema = z.object({
  name: z.string().min(1, "Name is required"),
  email: z.string().email().optional().nullish().or(z.literal("")),
  phone: z.string().optional().nullish().or(z.literal("")),
  location: z.string().optional().nullish().or(z.literal("")),
  portfolioUrl: z
    .string()
    .url("Invalid URL. URL should be of the format: 'https://www.abcd.com'")
    .optional()
    .nullish()
    .or(z.literal("")),
  githubUrl: z
    .string()
    .url("Invalid URL. URL should be of the format: 'https://www.abcd.com'")
    .optional()
    .nullish()
    .or(z.literal("")),
  linkedinUrl: z
    .string()
    .url("Invalid URL. URL should be of the format: 'https://www.abcd.com'")
    .optional()
    .nullish()
    .or(z.literal("")),
  stackoverflowUrl: z
    .string()
    .url("Invalid URL. URL should be of the format: 'https://www.abcd.com'")
    .optional()
    .nullish()
    .or(z.literal("")),
  blogUrl: z
    .string()
    .url("Invalid URL. URL should be of the format: 'https://www.abcd.com'")
    .optional()
    .nullish()
    .or(z.literal("")),
});

export const workExperienceFieldSchema = z.object({
  jobTitle: z.string().min(1, "You need to enter the job title"),
  jobSubtitle: z.string().optional().nullish().or(z.literal("")),
  companyName: z.string().optional().nullish().or(z.literal("")),
  location: z.string().optional().nullish().or(z.literal("")),
  duration: durationFieldSchema.optional().nullish(),
  details: z
    .string()
    .min(1, "You need to enter the description of your job experience"),
});

export const projectsFieldSchema = z.object({
  projectTitle: z
    .string()
    .min(1, "You need to enter the title of your project"),
  projectSubtitle: z.string().optional().nullish().or(z.literal("")),
  projectUrl: z.string().optional().nullish().or(z.literal("")),
  associatedWith: z.string().optional().nullish().or(z.literal("")),
  duration: durationFieldSchema.optional().nullish(),
  details: z.string().optional().nullish().or(z.literal("")),
});
export const skillsFieldSchema = z.object({
  title: z.string().optional().nullish().or(z.literal("")),
  skills: z.string().min(1, "This is a required field"),
});

export const educationFieldSchema = z.object({
  universityName: z.string().min(1, "University Name is required"),
  degreeName: z.string().min(1, "Degree Name is required"),
  majorName: z.string().optional().nullish().or(z.literal("")),
  grade: z.string().optional().nullish().or(z.literal("")),
  location: z.string().optional().nullish().or(z.literal("")),
  duration: durationFieldSchema.optional().nullish(),
});

export const basicDetailsSectionSchema = z.object({
  type: z.literal(SECTION.BASIC_DETAILS),
  sectionTitle: z.string().min(1, "The section title is required"),
  fields: basicDetailsFieldSchema,
});
export const professionalSummarySectionSchema = z.object({
  type: z.literal(SECTION.PROFESSIONAL_SUMMARY),
  sectionTitle: z.string().min(1, "The section title is required"),
  fields: z.object({
    value: z.string().min(1, "You can not leave this field as empty"),
  }),
});
export const workExperienceSectionSchema = z.object({
  type: z.literal(SECTION.WORK_EXPERIENCE),
  sectionTitle: z.string().min(1, "The section title is required"),
  fields: z.array(workExperienceFieldSchema),
});
export const projectsSectionSchema = z.object({
  type: z.literal(SECTION.PROJECTS),
  sectionTitle: z.string().min(1, "The section title is required"),
  fields: z.array(projectsFieldSchema),
});
export const skillsSectionSchema = z.object({
  type: z.literal(SECTION.SKILLS),
  sectionTitle: z.string().min(1, "The section title is required"),
  fields: z.array(skillsFieldSchema),
});
export const educationSectionSchema = z.object({
  type: z.literal(SECTION.EDUCATION),
  sectionTitle: z.string().min(1, "The section title is required"),
  fields: z.array(educationFieldSchema),
});
export const additionalSectionSchema = z.object({
  type: z.literal(SECTION.ADDITIONAL),
  sectionTitle: z.string().min(1, "The section title is required"),
  fields: z.object({
    value: z.string().min(1, "This information is required"),
  }),
});

export const sectionTypes = z.union([
  professionalSummarySectionSchema,
  workExperienceSectionSchema,
  projectsSectionSchema,
  skillsSectionSchema,
  educationSectionSchema,
  additionalSectionSchema,
]);

export const formSchema = z.object({
  basicDetails: basicDetailsSectionSchema,
  optionalSections: z.array(sectionTypes),
});

export type formType = z.infer<typeof formSchema>;

export const resumeVariantGenerationFormSchema = z.object({
  jobDescription: z
    .string()
    .min(
      1,
      "It looks like you have missed to add the job description. A job description is required for generating a variant of your resume."
    ),
  modifiedResumeJSON: z
    .string()
    .min(1, "Please enter the JSON response from chat GPT")
    .refine((data) => {
      return formSchema.safeParse(data).success;
    }, "The entered data is not in the correct format. Please try entering both the commands in a new chatGPT conversation. If the issue continues, please write to us"),
  customResumeName: z.string().optional().nullish().or(z.literal("")),
});
