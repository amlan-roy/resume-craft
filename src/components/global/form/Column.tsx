import React from "react";
import {
  SortableContext,
  verticalListSortingStrategy,
} from "@dnd-kit/sortable";
import Task from "./Task";

type ColumnProps = {
  tasks: { id: Number; title: String }[];
};

const Column: React.FC<ColumnProps> = ({ tasks }) => {
  return (
    <div className="bg-gray-200 rounded-sm w-4/5 max-w-lg flex flex-col gap-4 p-6">
      <SortableContext
        items={tasks.map((task) => ({ ...task, id: task.id.toString() }))}
        strategy={verticalListSortingStrategy}
      >
        {tasks.map((task) => (
          <Task id={task.id} title={task.title} key={task.id.toString()}></Task>
        ))}
      </SortableContext>
    </div>
  );
};
export default Column;
