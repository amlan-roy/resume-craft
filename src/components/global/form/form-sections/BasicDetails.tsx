import React from "react";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import HiddenInput from "@/components/global/form/form-inputs/HiddenInput";
import TextInput from "@/components/global/form/form-inputs/TextInput";
import { SECTION, formType } from "@/lib/types/form";
import { useFormContext } from "react-hook-form";

type BasicDetailsProps = {
  sectionTitle?: string;
  fieldName?: keyof formType;
};

const fieldName = "basicDetails";

/**
 * This is a compoenet used to render the Basic Details section in the resume data input form.
 * This is tightly coupled with the formSchema from zod, and any changes there might requrie changes here as well
 * (especially in the fieldName and error messages)
 */
const BasicDetails: React.FC<BasicDetailsProps> = ({
  sectionTitle = "Basic Details",
}) => {
  const {
    register,
    formState: { errors },
  } = useFormContext<formType>();

  return (
    <Card data-testid="form-sections__basic-details">
      <HiddenInput
        fieldName={fieldName && `${fieldName}.type`}
        value={SECTION.BASIC_DETAILS}
        register={register}
      />
      <CardHeader className="text-brand-neutral-11 text-3xl md:text-4xl">
        {sectionTitle && <CardTitle>{sectionTitle}</CardTitle>}
      </CardHeader>
      <CardContent className="flex flex-wrap w-full gap-5 justify-between">
        <HiddenInput
          fieldName={fieldName && `${fieldName}.sectionTitle`}
          value={sectionTitle}
          register={register}
        />
        <TextInput
          fieldName={fieldName && `${fieldName}.fields.name`}
          register={register}
          label="Name"
          autoComplete="name"
          placeholder="Stephen Hawwkins"
          className="w-full lg:max-w-[30%] md:max-w-[45%]"
          errorMessage={errors?.[fieldName]?.fields?.name?.message}
        />
        <TextInput
          fieldName={fieldName && `${fieldName}.fields.email`}
          register={register}
          label="Email"
          autoComplete="email"
          placeholder="little-people@complex-equations.com"
          className="w-full lg:max-w-[30%] md:max-w-[45%]"
          errorMessage={errors?.[fieldName]?.fields?.email?.message}
        />
        <TextInput
          fieldName={fieldName && `${fieldName}.fields.phone`}
          register={register}
          label="Phone Number"
          autoComplete="tel"
          placeholder="+69 4206996024"
          className="w-full lg:max-w-[30%] md:max-w-[45%]"
          errorMessage={errors?.[fieldName]?.fields?.phone?.message}
        />
        <TextInput
          fieldName={fieldName && `${fieldName}.fields.location`}
          register={register}
          label="Location"
          autoComplete="address-level2"
          placeholder="Epstein Island"
          className="w-full lg:max-w-[30%] md:max-w-[45%]"
          errorMessage={errors?.[fieldName]?.fields?.location?.message}
        />
        <TextInput
          fieldName={fieldName && `${fieldName}.fields.portfolioUrl`}
          register={register}
          label="Portfolio Url"
          autoComplete="url"
          placeholder="www.tooHighUpChalkboard.com"
          className="w-full lg:max-w-[30%] md:max-w-[45%]"
          errorMessage={errors?.[fieldName]?.fields?.portfolioUrl?.message}
        />
        <TextInput
          fieldName={fieldName && `${fieldName}.fields.githubUrl`}
          register={register}
          label="Github Url"
          autoComplete="url"
          placeholder="www.github.com/thisIsAShocker"
          className="w-full lg:max-w-[30%] md:max-w-[45%]"
          errorMessage={errors?.[fieldName]?.fields?.githubUrl?.message}
        />
        <TextInput
          fieldName={fieldName && `${fieldName}.fields.linkedinUrl`}
          register={register}
          label="LinkedIn Url"
          autoComplete="url"
          placeholder="www.linkedin.com/narakMeJaega"
          className="w-full lg:max-w-[30%] md:max-w-[45%]"
          errorMessage={errors?.[fieldName]?.fields?.linkedinUrl?.message}
        />
        <TextInput
          fieldName={fieldName && `${fieldName}.fields.stackoverflowUrl`}
          register={register}
          label="Stack Overflow Url"
          autoComplete="url"
          placeholder="www.stackoverflow.com/iAmOutOfIdeas"
          className="w-full lg:max-w-[30%] md:max-w-[45%]"
          errorMessage={errors?.[fieldName]?.fields?.stackoverflowUrl?.message}
        />
        <TextInput
          fieldName={fieldName && `${fieldName}.fields.blogUrl`}
          register={register}
          label="Blog Url"
          autoComplete="url"
          placeholder="www.cantFindFunnyPlaceholders.com"
          className="w-full lg:max-w-[30%] md:max-w-[45%]"
          errorMessage={errors?.[fieldName]?.fields?.blogUrl?.message}
        />
      </CardContent>
    </Card>
  );
};
export default BasicDetails;
