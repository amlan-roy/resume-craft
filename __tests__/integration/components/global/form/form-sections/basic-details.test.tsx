import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { SECTION } from "@/lib/types/form";
import BasicDetails from "@/components/global/form/form-sections/BasicDetails";

describe("integration:components/global/form/form-sections/basic-details", () => {
  it.skip("snapshot", () => {
    const tree = render(<BasicDetails sectionTitle="Section Title" />);

    expect(tree.container).toMatchSnapshot();
  });

  it.skip("BasicDetails component renders correctly", () => {
    render(
      <BasicDetails
        sectionTitle="Personal Information"
        register={jest.fn()}
        fieldName="basicDetails"
      />
    );
    const card = screen.getByTestId("form-sections__basic-details");
    expect(card).toBeInTheDocument();

    const title = screen.getByText("Personal Information");
    expect(title).toBeInTheDocument();

    const hiddenInput = card.querySelector(
      '[data-testid="hidden-input__container"]'
    );
    expect(hiddenInput).toHaveValue(SECTION.BASIC_DETAILS);

    const inputs = screen.getAllByTestId("text-input__container");

    // testing this way also tests the order of the inputs
    expect(inputs.length).toStrictEqual(9);

    expect(inputs[0].querySelector("label")).toHaveTextContent("Name");
    expect(inputs[0].querySelector("input")).toBeInTheDocument();
    expect(inputs[0].querySelector("p")).not.toBeInTheDocument();

    expect(inputs[1].querySelector("label")).toHaveTextContent("Email");
    expect(inputs[1].querySelector("input")).toBeInTheDocument();
    expect(inputs[1].querySelector("p")).not.toBeInTheDocument();

    expect(inputs[2].querySelector("label")).toHaveTextContent("Phone Number");
    expect(inputs[2].querySelector("input")).toBeInTheDocument();
    expect(inputs[2].querySelector("p")).not.toBeInTheDocument();

    expect(inputs[3].querySelector("label")).toHaveTextContent("Location");
    expect(inputs[3].querySelector("input")).toBeInTheDocument();
    expect(inputs[3].querySelector("p")).not.toBeInTheDocument();

    expect(inputs[4].querySelector("label")).toHaveTextContent("Portfolio Url");
    expect(inputs[4].querySelector("input")).toBeInTheDocument();
    expect(inputs[4].querySelector("p")).not.toBeInTheDocument();

    expect(inputs[5].querySelector("label")).toHaveTextContent("Github Url");
    expect(inputs[5].querySelector("input")).toBeInTheDocument();
    expect(inputs[5].querySelector("p")).not.toBeInTheDocument();

    expect(inputs[6].querySelector("label")).toHaveTextContent("LinkedIn Url");
    expect(inputs[6].querySelector("input")).toBeInTheDocument();
    expect(inputs[6].querySelector("p")).not.toBeInTheDocument();

    expect(inputs[7].querySelector("label")).toHaveTextContent(
      "Stack Overflow Url"
    );
    expect(inputs[7].querySelector("input")).toBeInTheDocument();
    expect(inputs[7].querySelector("p")).not.toBeInTheDocument();

    expect(inputs[8].querySelector("label")).toHaveTextContent("Blog Url");
    expect(inputs[8].querySelector("input")).toBeInTheDocument();
    expect(inputs[8].querySelector("p")).not.toBeInTheDocument();
  });

  // todo: Add tests to test error once user events issue is figured out
});
