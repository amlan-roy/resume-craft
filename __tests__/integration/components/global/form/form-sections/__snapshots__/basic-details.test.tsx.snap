// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration:components/global/form/form-sections/basic-details snapshot 1`] = `
<div>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
    data-testid="form-sections__basic-details"
  >
    <input
      class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
      data-testid="hidden-input__container"
      hidden=""
      type="hidden"
      value="BASIC_DETAILS"
    />
    <div
      class="flex flex-col space-y-1.5 p-6 text-brand-neutral-11 text-3xl md:text-4xl"
    >
      <h3
        class="text-2xl font-semibold leading-none tracking-tight"
      >
        Section Title
      </h3>
    </div>
    <div
      class="p-6 pt-0 flex flex-wrap w-full gap-5 justify-between"
    >
      <input
        class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
        data-testid="hidden-input__container"
        hidden=""
        type="hidden"
        value="Section Title"
      />
      <div
        class="w-full lg:max-w-[30%] md:max-w-[45%]"
        data-testid="text-input__container"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-lg md:text-xl text-brand-neutral-11"
        >
          Name
        </label>
        <input
          autocomplete="name"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 my-3"
          placeholder="Stephen Hawwkins"
        />
      </div>
      <div
        class="w-full lg:max-w-[30%] md:max-w-[45%]"
        data-testid="text-input__container"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-lg md:text-xl text-brand-neutral-11"
        >
          Email
        </label>
        <input
          autocomplete="email"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 my-3"
          placeholder="little-people@complex-equations.com"
        />
      </div>
      <div
        class="w-full lg:max-w-[30%] md:max-w-[45%]"
        data-testid="text-input__container"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-lg md:text-xl text-brand-neutral-11"
        >
          Phone Number
        </label>
        <input
          autocomplete="tel"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 my-3"
          placeholder="+69 4206996024"
        />
      </div>
      <div
        class="w-full lg:max-w-[30%] md:max-w-[45%]"
        data-testid="text-input__container"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-lg md:text-xl text-brand-neutral-11"
        >
          Location
        </label>
        <input
          autocomplete="address-level2"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 my-3"
          placeholder="Epstein Island"
        />
      </div>
      <div
        class="w-full lg:max-w-[30%] md:max-w-[45%]"
        data-testid="text-input__container"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-lg md:text-xl text-brand-neutral-11"
        >
          Portfolio Url
        </label>
        <input
          autocomplete="url"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 my-3"
          placeholder="www.tooHighUpChalkboard.com"
        />
      </div>
      <div
        class="w-full lg:max-w-[30%] md:max-w-[45%]"
        data-testid="text-input__container"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-lg md:text-xl text-brand-neutral-11"
        >
          Github Url
        </label>
        <input
          autocomplete="url"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 my-3"
          placeholder="www.github.com/thisIsAShocker"
        />
      </div>
      <div
        class="w-full lg:max-w-[30%] md:max-w-[45%]"
        data-testid="text-input__container"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-lg md:text-xl text-brand-neutral-11"
        >
          LinkedIn Url
        </label>
        <input
          autocomplete="url"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 my-3"
          placeholder="www.linkedin.com/narakMeJaega"
        />
      </div>
      <div
        class="w-full lg:max-w-[30%] md:max-w-[45%]"
        data-testid="text-input__container"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-lg md:text-xl text-brand-neutral-11"
        >
          Stack Overflow Url
        </label>
        <input
          autocomplete="url"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 my-3"
          placeholder="www.stackoverflow.com/iAmOutOfIdeas"
        />
      </div>
      <div
        class="w-full lg:max-w-[30%] md:max-w-[45%]"
        data-testid="text-input__container"
      >
        <label
          class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-lg md:text-xl text-brand-neutral-11"
        >
          Blog Url
        </label>
        <input
          autocomplete="url"
          class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 my-3"
          placeholder="www.cantFindFunnyPlaceholders.com"
        />
      </div>
    </div>
  </div>
</div>
`;
